### PR TITLE
[refactor] add importconfig-controller

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

File: pkg/controller/importconfig/manager.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		return err
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &clusterv1.ManagedCluster{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.Predicate(predicate.Funcs{
+			GenericFunc: func(e event.GenericEvent) bool { return false },
+			DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+			UpdateFunc:  func(e event.UpdateEvent) bool { return false },
```
Comment (skeeey) [inline pkg/controller/importconfig/manager.go:59]: image registry label

File: pkg/controller/importconfig/importconfig_controller.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+		if err != nil {
+			// this should not happen, if happened, panic here
+			panic(err)
+		}
+		raw := helpers.MustCreateAssetFromTemplate(file, template, config)
+		importYAML.WriteString(fmt.Sprintf("%s%s", constants.YamlSperator, string(raw)))
+	}
+
+	crdsV1beta1YAML := new(bytes.Buffer)
+	crdsV1veta1, err := manifestFiles.ReadFile(klusterletCrdsV1veta1File)
```
Comment (qiujian16) [inline pkg/controller/importconfig/importconfig_controller.go:217]: typo klusterletCrdsV1veta1File

**Pull request merged**